### PR TITLE
Add new document on canonical string encoding

### LINKED ISSUE #28: Spec: We need to decide on and document a canonical string encoding for our AST / parsed SQL statements
It seems like so far we are assuming lowercase string output for all AST components, which seems fine. Since we aren't exposing the AST to the outside world (yet), a canonical encoding (JSON?) of the tree state itself _isn't_ really required at this time... but it would be nice to spec and document the _string encoding_ and finalize that sooner rather than later.

I would propose we stick with lowercase string outputs. It _would_ be nice to have a canonical JSON representation as well, at least to help with future-proofing ourselves to a degree. In that case, I would propose we use lower-case, snake-case field names for all AST nodes and elements.

If we are happy with this, then we can update our spec with this information.

cc @sanderpick @brunocalza @jsign

### PR DESCRIPTION
**Pull request by carsonfarmer**
Fixes #28

### CODE REVIEW

**Comment by carsonfarmer** [comment]
What do you think of this @brunocalza ? Should address the linked issue we discussed previously?

File: specs/sql/Encoding.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Encoding
+
+As mentioned in the section on [Statement Types](#statement-types), the core Tableland SQL parser accepts a semicolon-separated list of statements, which are then parsed and evaluated according this Tableland SQL Specification. Internally, the statements are represented using an [abstract syntax tree](https://en.wikipedia.org/wiki/Abstract_syntax_tree) (AST). The internal representation of the nodes of this AST are outside the bounds of the Tableland SQL Specification, however, further details can be found in the [Go Tableland SQL Parser](https://github.com/tablelandnetwork/go-sqlparser) reference implementation.
+
+With the above caveat in mind, the Tableland SQL Specification does define a canonical string encoding of a set of (compliant) SQL statements that have passed through the Tableland SQL Parser (and have been represented via the Parser's AST). That is, this Specfication outlines — in general terms — the string encoding produced by parsing a set of Tableland SQL Specification compliant statements and re-encoding them into a canonical (string) format.
```
Comment (brunocalza) [inline specs/sql/Encoding.md:5]: typo: `Specfication`

File: specs/sql/Encoding.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Encoding
+
+As mentioned in the section on [Statement Types](#statement-types), the core Tableland SQL parser accepts a semicolon-separated list of statements, which are then parsed and evaluated according this Tableland SQL Specification. Internally, the statements are represented using an [abstract syntax tree](https://en.wikipedia.org/wiki/Abstract_syntax_tree) (AST). The internal representation of the nodes of this AST are outside the bounds of the Tableland SQL Specification, however, further details can be found in the [Go Tableland SQL Parser](https://github.com/tablelandnetwork/go-sqlparser) reference implementation.
```
Comment (brunocalza) [inline specs/sql/Encoding.md:3]: "which are then parsed and evaluated according this Tableland SQL Specification", maybe should be "which are then parsed and evaluated according to Tableland SQL Specification"? (`this` -> `to`)

Comment (brunocalza) [reply specs/sql/Encoding.md:3]: "AST are outside" -> "AST is outside" ?

**Review by brunocalza — APPROVED** [review]
I left some comments about a typo and some grammar issues. Apart from that, the content looks great!

**Review by brunocalza — APPROVED** [review]

**Pull request merged**